spec/unit/htty/cli/commands/path_set_spec.rb

```diff
@@ -67,6 +67,34 @@
         built = klass.build_for('x baz', :session => :another_session)
         built.should == nil
       end
+
+      it 'should correctly URL escape simple paths' do
+        built = klass.build_for('path b@r', :session => :a_session)
+        built.should be_instance_of(klass)
+        built.arguments.should == ['b%40r']
+        built.session.should   == :a_session
+      end
+
+      it 'should correctly URL escape multi-part paths' do
+        built = klass.build_for('path bar/baz', :session => :a_session)
+        built.should be_instance_of(klass)
+        built.arguments.should == ['bar/baz']
+        built.session.should   == :a_session
+      end
+
+      it 'should correctly URL escape multi-part paths containing special characters' do
+        built = klass.build_for('path b@r/ba{', :session => :a_session)
+        built.should be_instance_of(klass)
+        built.arguments.should == ['b%40r/ba%7B']
+        built.session.should   == :a_session
+      end
+
+      it 'should correctly URL escape multi-part paths containing escaped slashes' do
+        built = klass.build_for('path b@r/b%2Fz', :session => :a_session)
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.